packages/amplify-graphql-api-construct/src/amplify-graphql-definition.ts

```diff
@@ -88,11 +89,38 @@ export class AmplifyGraphqlDefinition {
     if (definitions.length === 1) {
       return definitions[0];
     }
+
+    // A strategy will be present multiple times in a given definition: once per model. We'll create a unique list per definition to ensure
+    // no reuse across definitions.
+    let combinedStrategyNames: string[] = [];
+    for (const definition of definitions) {
+      const definitionStrategyNames = new Set<string>();
+      for (const strategy of Object.values(definition.dataSourceStrategies)) {
+        if (!isSqlStrategy(strategy)) {
+          continue;
```

Comment:
This is a bit confusing. We can have non-SQL i.e DDB strategy in multiple definitions but not each SQL strategy must be in it's own definition. Wondering what the high level DX is for these cases.

Reply:
It's unfortunately a bit confusing, which you can see in the [tests](https://github.com/aws-amplify/amplify-category-api/pull/2109/files#diff-08f45b7d33ae7b9e339a473334693ec754019ffcbfd773556ec5cec56fbdddb9R259-R265). 

We can explore a cleaner DX, but since that is an additive change, I'd prefer to wait until we get some customer feedback on it; it's awfully easy to put "convenience" methods in place that end up being cumbersome (I'm lookin' at you, `SQLLambdaModelDataSourceStrategyFactory.fromCustomSqlFiles`!)